client/src/overview/subComponents/StyleSelector.jsx

```diff
@@ -1,18 +1,52 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { StyleIdxContext, StyleNameContext } from '../overview.jsx';
+
+function StyleSelector({ stylePics, currentProductStyle }) {
+  const StyleIdxContextData = useContext(StyleIdxContext);
+  const StyleNameContextData = useContext(StyleNameContext);
+  let currentStyleIdx;
+  let setCurrentStyleIdx;
+  let currentStyleName;
+  let setCurrentStyleName;
+  if (StyleIdxContextData !== undefined) {
```

Comment:
We might not want to check against `undefined` specifically. Possibly just `if (StyleIdxContextData)`